src/object.d

```diff
@@ -1751,6 +1751,59 @@ unittest
 }
 
 
+/**
+ * Convenience mixin for trivially sub-classing exceptions
+ *
+ * Even trivially sub-classing an exception involves writing boilerplate code
+ * for the constructor to correctly pass in the source file and line number the
+ * exception was thrown from. This mixin makes it easier.
+ *
+ * Example:
+ * ---
+ * class MeaCulpa: Exception
+ * {
+ *     mixin StandardExceptionConstructor;
+ * }
+ * assertThrown!MeaCulpa(enforce!MeaCulpa(false));
+ *
+ * try throw new MeaCulpa;
+ * catch (MeaCulpa e)
+ * {
+ *     assert(e.file == __FILE__);
+ *     assert(e.line == __LINE__ - 4);
+ * }
+ * ---
+ *
+ * Note that if you want to pass in special arguments to your exception, you
+ * obviously can't use this mixin and have to write the constructor manually.
+ */
+mixin template StandardExceptionConstructor()
+{
+    this(string msg, string file = __FILE__, size_t line = __LINE__)
+    {
+        super(msg, file, line);
+    }
```

Comment:
Well the next parameter is *not* intended to be used from user code as you can see at: http://dlang.org/phobos/object.html#.Exception: "The `next` parameter is used internally and should always be null when passed by user code", and likewise the second overload is only needed for convenience of providing the `next` right after the message without having to manually include `__FILE__` and `__LINE__`, so there's no point in including that overload either.

As for documentation, I thought using the mixin would actually enable it if you include a doc-comment block before it? Besides, if all you are doing is a trivial constructor for purposes of identification, then there should be no *need* for documentation since it's constructed in no way different than a regular exception.